pkgs/development/interpreters/python/cpython/2.7/importrename.patch

```diff
@@ -0,0 +1,80 @@
+diff --git a/Python/import.c b/Python/import.c
+index 397f485..0c71073 100644
+--- a/Python/import.c
++++ b/Python/import.c
+@@ -951,6 +951,7 @@ static void
+ write_compiled_module(PyCodeObject *co, char *cpathname, struct stat *srcstat, time_t mtime)
+ {
+     FILE *fp;
++    char *cpathname_tmp;
+ #ifdef MS_WINDOWS   /* since Windows uses different permissions  */
+     mode_t mode = srcstat->st_mode & ~S_IEXEC;
+     /* Issue #6074: We ensure user write access, so we can delete it later
+@@ -963,7 +964,20 @@ write_compiled_module(PyCodeObject *co, char *cpathname, struct stat *srcstat, t
+     mode_t mode = srcstat->st_mode & ~S_IXUSR & ~S_IXGRP & ~S_IXOTH;
+ #endif
+ 
+-    fp = open_exclusive(cpathname, mode);
++    /* Under POSIX, we first write to a tmp file and then take advantage
++       of atomic renaming. */
++
++    /* first construct the filename with a .tmp suffix */
++    cpathname_tmp = PyMem_MALLOC(strlen(cpathname) + 5);
++    if (cpathname_tmp == NULL) {
++        PyErr_Clear();
```

Comment:
Indeed, we need to clear the error here because this function is okay to fail. I updated my patch.

I didn't use `goto` because the original author preferred less control flow and I wanted to keep the original patch's spirit.